### PR TITLE
prepare 0.3.0

### PR DESCRIPTION
**Pull request by zachschuermann**
release `0.3.0` 🚀 

(don't merge for me please, I'll adjust the date in the changelog when I'm actually about to do the release)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,31 @@
...
+- Add `--limit` option to example `read-table-multi-threaded` [\#297](https://github.com/delta-incubator/delta-kernel-rs/pull/297)
+- FFI now built with cmake. Move to using the read-test example as an ffi-test. And building on macos. [\#288](https://github.com/delta-incubator/delta-kernel-rs/pull/288)
+- Golden table tests migrated from delta-spark/delta-kernel java [\#295](https://github.com/delta-incubator/delta-kernel-rs/pull/295)
+- Code coverage implemented via [cargo-llvm-cov](https://github.com/taiki-e/cargo-llvm-cov) and reported with [codecov](https://app.codecov.io/github/delta-incubator/delta-kernel-rs) [\#287](https://github.com/delta-incubator/delta-kernel-rs/pull/287)
+- All tests enabled to run in CI [\#284](https://github.com/delta-incubator/delta-kernel-rs/pull/284)
+- Updated DAT to 0.3 [\#290](https://github.com/delta-incubator/delta-kernel-rs/pull/290)
+
+**Fixed bugs:**
+
+- Read UTC timestamps as "UTC" instead of "+00:00" for timezone [\#295](https://github.com/delta-incubator/delta-kernel-rs/pull/295)
```
Comment (nicklan) [inline CHANGELOG.md:25]: nit: "Evaluate timestamps as..."

File: kernel/src/actions/deletion_vector.rs [hunk 2]
```diff
@@ -167,6 +167,8 @@ impl DeletionVectorDescriptor {
         }
     }
 
+    /// Materialize the row indexes of the deletion vector as a Vec<u64> in which each element
```
Comment (nicklan) [inline kernel/src/actions/deletion_vector.rs:170]: Generally I don't love making any "real" changes in these kinds of PRs. For now it's probably fine, but we should get in the habit of making these first

**Review by nicklan — APPROVED** [review]
lgtm with one nit!

File: kernel/src/actions/deletion_vector.rs (hunk 2 above)
Comment (zachschuermann) [reply kernel/src/actions/deletion_vector.rs:170]: Ah yea agree probably not the best spot for it - just noticed we missed some documentation earlier..

Comment (hntd187) [reply kernel/src/actions/deletion_vector.rs:170]: Agreed with Nick, but I appreciate you plugging the hole here

**Review by hntd187 — APPROVED** [review]
LGMT

**Review by rtyler — APPROVED** [review]

**Pull request merged**